### ISSUE TITLE
The EditText in edit state of SI details is not looking good.

### ISSUE THREAD

**Issue by EGOR-IND** [openMF/mobile-wallet#1183]
**Summary:**

The EditText in edit state of SI details is showing the heading same as the text present in box which is not good looking because a heading should be description of what should be entered in the box. 

**Steps to reproduce:**

1. Go to SI Details activity,
2. Press edit float button,
3. Observe the EditText.

**Expected behavior:**

The Heading of the EditText should be different.

**Observed behavior:**

Right now the heading is same as the text inside the edit box. 

**Device and Android version:**

VIVO Y83, Android Version = 8.1.1

**Screenshots:**

<img src="https://user-images.githubusercontent.com/56648862/106017015-08b8b780-60e6-11eb-95e8-2b746a7105ce.jpeg" width="200" height="400"/>

**Comment by EGOR-IND** [comment]
I'm working on this one.

**Comment by divyank00** [comment]
The heading is same, to let the user know the previous value if he wishes to update it

**Comment by divyank00** [comment]
I don't think this is an issue :-)

**Comment by EGOR-IND** [comment]
> The heading is same, to let the user know the previous value if he wishes to update it

But i think user can see the amount in the box too i can show you some articles how these headings are used, and i think this heading is not intentional because it is using the same widget to display the amount. So i think it's a good enhancement by the way i also don't take it as an issue it is an enhancement, but devansh told me to make a issue for every PR.

**Comment by divyank00** [comment]
Assume you are updating values...You change the value in edit text. Now you feel no this value is not corrext i should revert back, so can you do? Thats why the previous value is shown to user.

**Comment by EGOR-IND** [comment]
> Assume you are uodating values...You change the value in edit text. Now you feel no this value is not corrext i should revert back, so can you do? Thats why the previous value is shown to user.

If the user feels that the value is not correct then he can simply revert back by pressing back and if you want that the previous amount should be present there then it can be shown there in other widget, i'm just saying that heading is not good place for that. Is it good if we show the previous amount in other widget?

**Comment by EGOR-IND** [comment]
> > Assume you are uodating values...You change the value in edit text. Now you feel no this value is not corrext i should revert back, so can you do? Thats why the previous value is shown to user.
> 
> If the user feels that the value is not correct then he can simply revert back by pressing back and if you want that the previous amount should be present there then it can be shown there in other widget, i'm just saying that heading is not good place for that. Is it good if we show the previous amount in other widget?

Hey @divyank00 did this resolved your doubts about what I want this to look like. Please reply if you still have any doubt because brainstorming is good:).

**Comment by divyank00** [comment]
> > Assume you are uodating values...You change the value in edit text. Now you feel no this value is not corrext i should revert back, so can you do? Thats why the previous value is shown to user.
> 
> If the user feels that the value is not correct then he can simply revert back by pressing back and if you want that the previous amount should be present there then it can be shown there in other widget, i'm just saying that heading is not good place for that. Is it good if we show the previous amount in other widget?

there are 2 values that can be modiefied by the user, he changes two values then reverts one. 

I felt let the mentors decide what UI is expected

**Comment by EGOR-IND** [comment]
> > > Assume you are uodating values...You change the value in edit text. Now you feel no this value is not corrext i should revert back, so can you do? Thats why the previous value is shown to user.
> > 
> > 
> > If the user feels that the value is not correct then he can simply revert back by pressing back and if you want that the previous amount should be present there then it can be shown there in other widget, i'm just saying that heading is not good place for that. Is it good if we show the previous amount in other widget?
> 
> there are 2 values that can be modiefied by the user, he changes two values then reverts one.
> 
> I felt let the mentors decide what UI is expected

That's what I'm saying we can add another widget for previous amount and give name to the heading like previous and new. @devansh-299  what are your thoughts about this.

**Comment by devansh-299** [comment]
> 
> 
> > > > Assume you are uodating values...You change the value in edit text. Now you feel no this value is not corrext i should revert back, so can you do? Thats why the previous value is shown to user.
> > > 
> > > 
> > > If the user feels that the value is not correct then he can simply revert back by pressing back and if you want that the previous amount should be present there then it can be shown there in other widget, i'm just saying that heading is not good place for that. Is it good if we show the previous amount in other widget?
> > 
> > 
> > there are 2 values that can be modiefied by the user, he changes two values then reverts one.
> > I felt let the mentors decide what UI is expected
> 
> That's what I'm saying we can add another widget for previous amount and give name to the heading like previous and new. @devansh-299 what are your thoughts about this.

I left the heading same as the previous text to let the users know the last value as pointed out by @divyank00.  If you have any suggestions please elaborate so that we can together come with a better design. Let's use this thread only for this discussion!

**Comment by EGOR-IND** [comment]
> > > > > Assume you are uodating values...You change the value in edit text. Now you feel no this value is not corrext i should revert back, so can you do? Thats why the previous value is shown to user.
> > > > 
> > > > 
> > > > If the user feels that the value is not correct then he can simply revert back by pressing back and if you want that the previous amount should be present there then it can be shown there in other widget, i'm just saying that heading is not good place for that. Is it good if we show the previous amount in other widget?
> > > 
> > > 
> > > there are 2 values that can be modiefied by the user, he changes two values then reverts one.
> > > I felt let the mentors decide what UI is expected
> > 
> > 
> > That's what I'm saying we can add another widget for previous amount and give name to the heading like previous and new. @devansh-299 what are your thoughts about this.
> 
> I left the heading same as the previous text to let the users know the last value as pointed out by @divyank00. If you have any suggestions please elaborate so that we can together come with a better design. Let's use this thread only for this discussion!

Yeah actually i think that we can show a text view for previous amount or maybe more better that we can make a dialog box for previous and new values, actually If I talk on more broad scale than I must say that I was thinking about changing the structure of SI details activity, I think that you also think that SI details needs some enhancement UI wise. So, It's on you if you say thay we should enhance the activity than we can do that or we can enhance that editing part only or we can close this issue😊.

**Comment by deepak-prajapatii** [comment]
@devansh-299 @EGOR-IND Can I work on this issue?

**Comment by EGOR-IND** [comment]
> @devansh-299 @EGOR-IND Can I work on this issue?

@deepak-prajapatii actually this issue is under discussion, so it is not finalized yet that these changes are needed or not, so at this moment this issue doesn't need any PR, once we finalize that what should be done than you can ask about making a PR😊.

**Comment by deepak-prajapatii** [comment]
> > @devansh-299 @EGOR-IND Can I work on this issue?
> 
> @deepak-prajapatii actually this issue is under discussion, so it is not finalized yet that these changes are needed or not, so at this moment this issue doesn't need any PR, once we finalize that what should be done than you can ask about making a PR😊.

@EGOR-IND Thanks for your confirmation. I will look for another one.

**Comment by EGOR-IND** [comment]
> > > @devansh-299 @EGOR-IND Can I work on this issue?
> > 
> > 
> > @deepak-prajapatii actually this issue is under discussion, so it is not finalized yet that these changes are needed or not, so at this moment this issue doesn't need any PR, once we finalize that what should be done than you can ask about making a PR😊.
> 
> @EGOR-IND Thanks for your confirmation. I will look for another one.

Hey @deepak-prajapatii, you can make a issue by yourself if you see any.

**Comment by deepak-prajapatii** [comment]
> > > > @devansh-299 @EGOR-IND Can I work on this issue?
> > > 
> > > 
> > > @deepak-prajapatii actually this issue is under discussion, so it is not finalized yet that these changes are needed or not, so at this moment this issue doesn't need any PR, once we finalize that what should be done than you can ask about making a PR😊.
> > 
> > 
> > @EGOR-IND Thanks for your confirmation. I will look for another one.
> 
> Hey @deepak-prajapatii, you can make a issue by yourself if you see any.

@EGOR-IND okay, I will try

**Comment by therajanmaurya** [comment]
Closing as we moved purely on compose.
